### DISCUSSION TITLE
How to enforce using Math.Round in the generated SQL

### DISCUSSION THREAD

**Discussion by yallie** [linq2db/linq2db#3426] [Q&A] [answered]
Consider a query

```csharp
from t in MyTable
select new
{
    Amount = Math.Round(t.Amount, 6)
}
```

It generates SQL like that:

```SQL
SELECT T.AMOUNT
FROM MY_TABLE T
```

Looks like `Math.Round` is executed on the client-side during mapping.
I'd like it to be like this instead, but can't get it to work:

```SQL
SELECT ROUND(T.AMOUNT, 6)
FROM MY_TABLE T
```

Column MY_TABLE.AMOUNT is numeric and is mapped to .NET decimal type.
Sometimes it has values that cannot be represented as .NET decimals.
They cause Npgsql data reader to throw `OverflowException`.
I'd like to avoid feeding the original values to the data reader.

**Comment by jods4** [comment, marked as answer]
Wrapping expressions with `AsSql` forces them to be executed server-side I believe?

**Reply by yallie** [reply]
Thanks! Exactly what I needed!

```csharp
from t in MyTable
select new
{
    Amount = Sql.AsSql(Math.Round(t.Amount, 6))
}
```
